multipart-post.lua

```diff
@@ -36,44 +40,111 @@ local gen_boundary = function()
   return table.concat(t)
 end
 
-local encode = function(t, boundary)
```

Comment:
This breaks the use of just `encode` and not `gen_request` (see README) which is how a lot of people use this module. My bad for not adding a test for that...

I agree that the ability to send large files is important but we should retain that possibility and not break backwards compatibility (this module is 7 years old and used in many places).

I propose something like [this](https://gist.github.com/catwell/d9845d3bd5ab4f5375c6a36845c0b0f5). Thoughts?